modules/apps/portal-configuration/portal-configuration-cluster/src/main/java/com/liferay/portal/configuration/cluster/internal/messaging/ConfigurationMessageListener.java

```diff
@@ -51,16 +51,15 @@ public void setReloadablePersistenceManager(
 
 	@Override
 	protected void doReceive(Message message) throws Exception {
-		if (message.contains(ConfigurationAdmin.SERVICE_FACTORYPID)) {
+		if (message.contains(Constants.SERVICE_PID)) {
 			reloadConfiguration(
-				message.getString(ConfigurationAdmin.SERVICE_FACTORYPID),
-				ConfigurationAdmin.SERVICE_FACTORYPID,
+				message.getString(Constants.SERVICE_PID), Constants.SERVICE_PID,
 				message.getInteger("configuration.event.type"));
 		}
-
-		if (message.contains(Constants.SERVICE_PID)) {
+		else if (message.contains(ConfigurationAdmin.SERVICE_FACTORYPID)) {
```

Comment:
Correct me if I'm wrong about this, but I think this second block is now unreachable.  The only place where I see we  actually send this message is from `ConfigurationSynchronousConfigurationListener`.  Since we _always_ set a PID but _conditionally_ set a factory PID, we will always enter this first block, which makes the second block effectively useless. See https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-configuration/portal-configuration-cluster/src/main/java/com/liferay/portal/configuration/cluster/internal/ConfigurationSynchronousConfigurationListener.java#L50-L54.  Am I missing something here?

Reply:
@jrao @ChrisKian

Reply:
@drewbrokke No, you're right. I just found https://issues.liferay.com/browse/LPS-69059 which I believe introduced the bug I'm trying to fix here. I'll revisit this and try to make sure I'm addressing both issues.